discord/bot.py

```diff
@@ -577,8 +577,8 @@ async def sync_commands(
             guild_permissions: List = []
 
             for i in commands:
-                cmd = find(lambda cmd: cmd.name == i["name"] and cmd.type == i["type"] and cmd.guild_ids is not None
-                                       and (i["guild_id"]) in cmd.guild_ids, self.pending_application_commands)
+                cmd = find(lambda cmd: cmd.name == i["name"] and str(cmd.type) == str(i["type"]) and cmd.guild_ids is not None
```

Comment:
Aren't `cmd.type` and `i["type"]` both integers?
```suggestion
                cmd = find(lambda cmd: cmd.name == i["name"] and cmd.type == i["type"] and cmd.guild_ids is not None
```

Reply:
Yes, but I didn't want to take any chances, sorry in advance :)

Reply:
No need to be sorry, it's fine if other devs approve this. I just want to avoid unnecessary conversions to keep the speed.

Reply:
I think @wurgo's change is fine.

Reply:
I'm struggling to see a situation where cmd.type and i["type"] would ever not be integers based on the latest command registration rewrite, and agree with @Dorukyum that we should avoid unnecessary conversions where possible.